include/libpmemfile-posix.h

```diff
@@ -158,9 +157,47 @@ extern "C" {
 typedef struct pmemfilepool PMEMfilepool;
 typedef struct pmemfile_file PMEMfile;
 
+typedef unsigned pmemfile_mode_t;
+typedef unsigned pmemfile_uid_t;
+typedef unsigned pmemfile_gid_t;
+typedef long long pmemfile_ssize_t;
+typedef long long pmemfile_off_t;
+typedef unsigned long pmemfile_nlink_t;
+typedef long pmemfile_blksize_t;
+typedef long pmemfile_blkcnt_t;
+typedef unsigned long pmemfile_dev_t;
+typedef unsigned long pmemfile_ino_t;
+
+struct pmemfile_timespec
+{
+	long tv_sec;
+	long tv_nsec;
+};
+
+struct pmemfile_stat
+{
+	pmemfile_dev_t st_dev;
+	pmemfile_ino_t st_ino;
+	pmemfile_nlink_t st_nlink;
```

Comment:
Is this typedef justified now?
Can't we just say:
'unsigned st_nlink' ?
Since these all are just integers, the compiler doesn't really catch errors where an argument of wrong type is passed to a function. Theoretically, 'unsigned' only guarantees an at least 16 bits wide integer, and what if a file has 70000 links...
Anyways, perhaps we don't have to prepare for any exotic scenario that POSIX prepares for.